### PR TITLE
fix bug in batched_forward_pass

### PR DESCRIPTION
**Pull request by ArvinZhuang**
fix the bug that will cause the devices not match issue in the `batched_forward_pass` method.

The reason:
`self. data_collator ` returned tensors will be on CPU, thus the later `self.model(**input_kwargs)` will give error as model is on GPU.

The proposed solution:
do `.to(self.accelerator.device)` after `self.data_collator`

My transformers version: 4.26.1

### CODE REVIEW

File: trl/trainer/ppo_trainer.py [hunk 1]
```diff
@@ -553,13 +553,12 @@ def batched_forward_pass(
         # automatically padded by the datacollator to indicate padding tokens
         if self.is_encoder_decoder:
             input_data = self.data_collator(
-                [{"input_ids": q, "attention_mask": torch.ones_like(q).to(self.accelerator.device)} for q in queries]
-            )
+                [{"input_ids": q, "attention_mask": torch.ones_like(q)} for q in queries]
+            ).to(self.accelerator.device)
 
-            attention_mask = [torch.ones_like(r) for r in responses]
```
Comment (younesbelkada) [inline trl/trainer/ppo_trainer.py:559]: Hum why this has been removed? 🤔

**Review by younesbelkada — COMMENTED** [review]
Wow thanks a lot for fixing the bug! 
I am ok with this fix in the principle that it is a safety checker to make sure all the returned tensors will be on the correct device (regardless where the dataloader will send the device)
Let's run the tests and see! 
Wdyt @lvwerra ?

File: trl/trainer/ppo_trainer.py (hunk 1 above)
Comment (ArvinZhuang) [reply trl/trainer/ppo_trainer.py:559]: because seems attention_mask variable is never used

Comment (younesbelkada) [reply trl/trainer/ppo_trainer.py:559]: Perfect, can you run the styling checks so that the testing suite will be executed?

Comment (ArvinZhuang) [reply trl/trainer/ppo_trainer.py:559]: just did

**Review by younesbelkada — COMMENTED** [review]
Thanks for fixing! Could you revert your changes in all the scripts inside `examples/` after that we should be good to merge!

**Comment by ArvinZhuang** [comment]
> Thanks for fixing! Could you revert your changes in all the scripts inside `examples/` after that we should be good to merge!
Hi 
these changes are automatically generated by style && quality, but I reverted them anyway

**Review by younesbelkada — APPROVED** [review]
Thanks for iterating! 🚀

**Comment by ArvinZhuang** [comment]
Also I notice this `"Forward batch size > 1 is not well supported yet for encoder-decoder models`.
Could you please give me more hits why this is the case? Looks like forward batch size =1 will greatly slow down the inference, I probably can give it a shot to fix this.

**Comment by Rebecca-Qian** [comment]
Had just prepared a PR to fix this 😄 thanks @ArvinZhuang

**Comment by ArvinZhuang** [comment]
> Had just prepared a PR to fix this 😄 thanks @ArvinZhuang
@Rebecca-Qian You are welcome :) Im glad to contribute as well

**Review by lvwerra — APPROVED** [review]
Thanks for the fix!

**Pull request merged**